src/SecureStore.BeyondTrust/UiPath.Orchestrator.BeyondTrust-DynamicSystem-ReadOnly/BeyondTrustDynamicSystemSecureStore.cs

```diff
@@ -0,0 +1,144 @@
+    {
+        public void Initialize(Dictionary<string, string> hostSettings)
+        {
+            // No host level settings
+        }
+
+        public Task ValidateContextAsync(string context)
+        {
+            BeyondTrustVaultClientFactory.GetClient(context).TestConnection();
+            return Task.CompletedTask;
+        }
+
+        public IEnumerable<ConfigurationEntry> GetConfiguration() => new List<ConfigurationEntry>
+            {
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = "Hostname",
+                    DisplayName = "BeyondTrust Host URL",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = "AuthKey",
+                    DisplayName = "API Authentication Key",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = "RunAs",
+                    DisplayName = "API Run As",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.Boolean)
+                {
+                    Key = "SSLEnabled",
+                    DisplayName = "Use SSL certificate",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = "SystemAccountDelimiter",
+                    DisplayName = "System-Account Delimiter",
+                    IsMandatory = true,
+                    DefaultValue = "/",
+                },
+                new ConfigurationValue(ConfigurationValueType.Choice)
+                {
+                    Key = "ManagedAccountType",
+                    DisplayName = "Managed Account Type",
+                    IsMandatory = true,
+                    PossibleValues = new List<string> { "system", "domainlinked" }
+                },
+            };
+
+        public SecureStoreInfo GetStoreInfo()
+        {
+            return new SecureStoreInfo { Identifier = "BeyondTrust-DynamicSystem-ReadOnly", IsReadOnly = true };
+        }
+
+        public async Task<string> CreateCredentialsAsync(string context, string key, Credential value)
+        {
+            throw new SecureStoreException(SecureStoreException.Type.UnsupportedOperation, "This secure store is read-only");
+        }
+
+        public async Task<Credential> GetCredentialsAsync(string context, string key)
+        {
+            var config = JsonConvert.DeserializeObject<Dictionary<string, object>>(context);
+            var splitDelimiter = new string[] { config["SystemAccountDelimiter"].ToString() };
+            var keyPieces = key.Split(splitDelimiter, StringSplitOptions.None);
+
+            if (keyPieces.Length > 2)
```

Comment:
What if there is just one part, because the delimiter is not found in the string? Consider using `!= 2`

Reply:
Fixed